image/roxctl/Dockerfile.rhtap

```diff
@@ -0,0 +1,53 @@
+# TODO: can we follow tags?
+FROM registry.access.redhat.com/ubi8/go-toolset:1.19 as builder
+
+WORKDIR /go/src/github.com/stackrox/rox/app
+
+# TODO: do we want to avoid this by COPYing each required directory separately?
+COPY . .
+
+RUN git config --global --add safe.directory /go/src/github.com/stackrox/rox/app && \
+    # We could add tag fetching to: https://github.com/redhat-appstudio/build-definitions/blob/main/task/git-clone/0.1/git-clone.yaml
+    # TODO: if git fetch fails, check if privileges/file owners are OK, usually only on local build because repo might have moved further.
+    # TODO: this prevents hermetic builds
+    git fetch --tags --force && \
+    mkdir -p image/bin
+
+ENV CI=1 GOFLAGS="" GOTAGS="release"
+
+RUN RACE=0 CGO_ENABLED=1 GOOS=linux GOARCH=$(go env GOARCH) BUILD_TAG=$(make tag) scripts/go-build.sh ./roxctl && \
+    cp bin/linux_$(go env GOARCH)/roxctl image/bin/roxctl
+
+# TODO: can we follow tags?
+FROM registry.access.redhat.com/ubi8/ubi-minimal:latest as app
+
+COPY --from=builder /go/src/github.com/stackrox/rox/app/image/bin/roxctl /usr/bin/roxctl
+
+# TODO: this prevents hermetic builds
+RUN microdnf upgrade -y --nobest && \
+    microdnf clean all && \
+    rpm --verbose -e --nodeps $(rpm -qa curl '*rpm*' '*dnf*' '*libsolv*' '*hawkey*' 'yum*') && \
+    rm -rf /var/cache/dnf /var/cache/yum
+
+LABEL \
+    com.redhat.component="rhacs-roxctl-container" \
+    name="rhacs-roxctl-rhel8" \
+    maintainer="Red Hat, Inc." \
+    # These labels are added to override the base image values.
+    description="The CLI for ACS" \
+    io.k8s.description="The CLI for ACS" \
+    io.k8s.display-name="roxctl" \
+    io.openshift.tags="rhacs,roxctl" \
+    summary="The CLI for ACS"
+
+    # TODO: what are these for & if they're required, how can we ?
+    # version="${CI_VERSION}" \
```

Comment:
`version` label is what drives the ultimate image tag (at least, its prefix before `-<build-number>`) for CPaaS-built containers. I think, we don't need to replicate it here at least for now.

Reply:
Removed label

Reply:
Unresolved because I checked the resulting container labels (used `docker inspect quay.io/redhat-user-workloads/rh-acs-tenant/acs/roxctl:on-pr-fdc1927c678083abd27a04736fdd073931a17c06`) and saw that we inherit label from base UBI stage. Not nice.

Since labels are taken from Dockerfile `ARG`-s and we don't have ability to customize `ARG`-s, I suggest for now we set dummy version to (a) override UBI's one, (b) signal it's not a valid one.

I believe, there's no possibility to run any command and use it's output in `LABEL`. Therefore, it should be solved by RHTAP. Let me start a thread.

Reply:
https://redhat-internal.slack.com/archives/C04PZ7H0VA8/p1697127151309229